### PR TITLE
refactor phone method availability checks

### LINKED ISSUE #663: phone methods remain in registry after taking out 'two_factor.plugins.phonenumber' from INSTALLED_APPS
## Expected Behavior
`two_factor.plugins.phonenumber.app.register_methods` should unregister `PhoneCallMethod` and `SMSMethod` if the phone number plug-in is removed.

## Current Behavior
They stay registered after removing the phone number plug-in.

### LINKED ISSUE #665: computing phone method availability is done in different places using different criteria
## Expected Behavior
There should be a single way to check which phone methods are available.

## Current Behavior
In some places availability is determined by the presence of `'two_factor.plugins.phonenumber'` in `INSTALLED_APPS`.
Other places take into account also / only values for `settings.TWO_FACTOR_CALL_GATEWAY` and `settings.TWO_FACTOR_SMS_GATEWAY`.

## Possible Solution
We already have a method registry, which should be the only source of truth.

### PR DESCRIPTION
**Pull request by jpaniagualaconich**
Fixes #663, fixes #665.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by claudep** [comment]
Would it be possible to move the TWO_FACTOR_SMS/CALL_GATEWAY settings to base test settings in a separate commit/PR?

**Comment by jpaniagualaconich** [comment]
@claudep ok, dropped that part.

File: two_factor/plugins/phonenumber/apps.py [hunk 1]
```diff
@@ -12,19 +12,22 @@ class TwoFactorPhoneNumberConfig(AppConfig):
...
 
 
-def register_methods(sender, setting, value, **kwargs):
+def update_registered_methods(sender, setting, value, **kwargs):
     # This allows for dynamic registration, typically when testing.
     from .method import PhoneCallMethod, SMSMethod
 
-    if getattr(settings, 'TWO_FACTOR_CALL_GATEWAY', None):
+    installed_apps = getattr(settings, 'INSTALLED_APPS') or []
+    phone_number_app_installed = 'two_factor.plugins.phonenumber' in installed_apps
```
Comment (claudep) [inline two_factor/plugins/phonenumber/apps.py:24]: This method of checking apps presence is brittle (as people may use custom app config which will be named differently). Do you think possible using `apps.is_installed()` instead?

Comment (jpaniagualaconich) [reply two_factor/plugins/phonenumber/apps.py]: ok, done!

**Review by claudep — APPROVED** [review]
Thanks, looks good!

**Pull request merged**